### DISCUSSION TITLE
How can I get the rate of production of all elementary reactions?

### DISCUSSION THREAD

**Discussion by Alex-jian522** [AMReX-Combustion/PeleLMeX#431] [General]
Hi everyone, I used a chemical reaction mechanism with 25 species  and 110 reactions. How can I get the rate of reactions of all 110 elementary reactions? Thank you very much for your answer

**Comment by baperry2** [comment]
A similar question was answered here: https://github.com/AMReX-Combustion/PelePhysics/discussions/511#discussioncomment-9961695

**Reply by Alex-jian522** [reply]
Thanks a lot. I will try to use Cantera.
![image](https://github.com/user-attachments/assets/623e7edc-ff2b-421f-9a40-9fbf3853feb0)
Another question is, you mentioned in your answer that Cantera needs pressure to calculate the rate of reactions, but the only variable saved is avg_pressure and the value is very small (as shown in the figure). Is the value of avg_pressure here the absolute deviation from the 1atm pressure I set?

**Reply by baperry2** [reply]
Yeah, `avg_pressure` is the perturbational pressure. The thermodynamic pressure that you would give to Cantera is nominally the 1 atm that you set, though it might be a bit more accurate to use the `RhoRT` variable. `RhoRT` is the actual thermodynamic pressure, which may deviate slightly from the nominal value if your SDC iterations have not converged.

**Reply by Alex-jian522** [reply]
Thank you very much for your clear and accurate reply.
I noticed that I saved the actual thermodynamic pressure RhoRT, but I didn't realize what it meant before. Therefore, I will use RhoRT in Cantera
